### PR TITLE
tests: Allow nested tests

### PR DESCRIPTION
**Pull request by johannes-wolf**
Tests are now detected by looking for files named "test.typ", rather than using each directory under "$TEST_ROOT".

### CODE REVIEW

File: tests/plot/parametric/test.typ [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#set page(width: auto, height: auto)
+#import "../../../src/lib.typ": *
```
Comment (fenjalien) [inline tests/plot/parametric/test.typ:2]: At this point would it be easier to use `/src/lib.typ`?

File: tests/plot/parametric/test.typ [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+             domain: (0, 2 * calc.pi), fill: true)
+    plot.add((t) => (calc.cos(t * 1rad) - .5, calc.sin(t * 1rad)),
+             domain: (0, 2 * calc.pi), fill: true)
+    plot.add((t) => (calc.cos(t * 1rad), calc.sin(t * 1rad) + .5),
+             domain: (0, 2 * calc.pi), fill: true)
+    plot.add((t) => (calc.cos(t * 1rad), calc.sin(t * 1rad) - .5),
+             domain: (0, 2 * calc.pi), fill: true)
+  })
+}))
+*/
```
Comment (fenjalien) [inline tests/plot/parametric/test.typ:73]: Why is this commented out?

File: tests/plot/parametric/test.typ [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+
+/* oo */
```
Comment (fenjalien) [inline tests/plot/parametric/test.typ:76]: Can this be a bit more descriptive?

**Review by fenjalien — CHANGES_REQUESTED** [review]
I'm impressed by how little the test script needed to change. It looks like you need to update the reference images though

File: tests/plot/parametric/test.typ (hunk 3 above)
Comment (johannes-wolf) [reply tests/plot/parametric/test.typ]: I removed it. It gets re-added in the contour PR.

File: tests/plot/parametric/test.typ (hunk 2 above)
Comment (johannes-wolf) [reply tests/plot/parametric/test.typ]: I removed it. It gets re-added in the contour PR as it renders wrong in this version.

**Review by fenjalien — APPROVED** [review]

**Pull request merged**